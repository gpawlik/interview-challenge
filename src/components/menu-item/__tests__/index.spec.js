import * as React from "react";
import renderer from "react-test-renderer";

import { MenuItem } from "../";

jest.mock("../styles", () => ({
  Container: "Container",
  Title: "Title",
  InfoRow: "InfoRow",
  RemoveButton: "RemoveButton",
}));

const initialProps = {
  name: "Item 123",
  dietaries: ["ab", "cd", "ef"],
  onPress: () => {},
};

const mockComponent = (props) => {
  const finalProps = {
    ...initialProps,
    ...props,
  };

  return renderer.create(<MenuItem {...finalProps} />);
};

describe("MenuItem", () => {
  it("renders correctly default version", () => {
    const tree = mockComponent().toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with delete button", () => {
    const tree = mockComponent({ onDelete: () => {} }).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
