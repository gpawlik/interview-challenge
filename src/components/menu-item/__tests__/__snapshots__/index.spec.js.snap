// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItem renders correctly default version 1`] = `
<Container
  onClick={[Function]}
>
  <Title>
    Item 123
  </Title>
  <InfoRow>
    <div
      className="sc-bdVaJa RUiaz"
    >
      ab
    </div>
    <div
      className="sc-bdVaJa RUiaz"
    >
      cd
    </div>
    <div
      className="sc-bdVaJa RUiaz"
    >
      ef
    </div>
  </InfoRow>
</Container>
`;

exports[`MenuItem renders correctly with delete button 1`] = `
<Container
  onClick={[Function]}
>
  <Title>
    Item 123
  </Title>
  <InfoRow>
    <div
      className="sc-bdVaJa RUiaz"
    >
      ab
    </div>
    <div
      className="sc-bdVaJa RUiaz"
    >
      cd
    </div>
    <div
      className="sc-bdVaJa RUiaz"
    >
      ef
    </div>
  </InfoRow>
  <RemoveButton
    onClick={[Function]}
  >
    x
  </RemoveButton>
</Container>
`;
